src/relocation.rs

```diff
@@ -237,15 +222,8 @@ mod overrides {
                 let mut map = map.borrow_mut();
 
                 for prefix in self.prefixes.drain() {
-                    match map.entry(prefix) {
-                        Entry::Occupied(mut entry) => {
-                            if entry.get().rc <= 1 {
-                                let _ = entry.remove();
-                            } else {
-                                entry.get_mut().rc -= 1;
-                            }
-                        }
-                        Entry::Vacant(_) => panic!("Unexpected missing overrides for {:?}", prefix),
+                    if let Some(info) = map.get_mut(&prefix) {
+                        info.next = None;
```

Comment:
Yes, because we no longer clear the already assigned mappings of original to overridden names. We only clear the future overrides (that a node from some given prefix P gets relocated to a name from some other given prefix Q). Because of that we no longer need to track the reference counter.